cli/src/commands/git/clone.rs

```diff
@@ -144,6 +144,12 @@ pub fn cmd_git_clone(
             .ok();
         }
     }
+    if args.colocate {
+        writeln!(
+            ui.warning_default(),
+            r#"Running `git clean -xdf` will remove `.jj/`!"#,
+        )?;
+    }
```

Comment:
nit: print at the end of the command because the warning isn't useful if `clone` failed.

Reply:
Beware that `clone_result` isn't unwrapped yet. I also think it's better to print the hint after the checkout information because it is a hint for future user action.

Reply:
I put it before the working-copy messages on purpose but I don't feel strongly so I've moved it last.